### DISCUSSION TITLE
Loading different SCSS files depend on layout component.

### DISCUSSION THREAD

**Discussion by mena234** [vercel/next.js#39316] [Help]
I have two layout components, let's call them Layout A and Layout B.

Each layout has a separate SCSS style and importing both together in the `_app.tsx` component will make conflicts.

I know which layout I'm in by URL PATH.

So, let's say 

{root_url}/layout_a/something   =>  Will show layout A
{root_url}/layout_b/something   => Will show layout B

**Just note each layout contains a lot of pages but this is a simple example.**

I tried a lot of ways to do it by ( dynamic load SCSS files ) but no success.

[I found this answer it seems work with React.js but not with Next.js](https://stackoverflow.com/questions/66531412/dynamically-load-and-unload-content-of-css-file-in-javascript-react)

When I try to use `import { default as Theme } from '!css-loader!sass-loader!../file/path'`, to disable webpack style loader, but it gives me an error "Can't resolve 'css-loader'.

I think this is a popular use case, I don't believe that there is no documentation for it.
Any idea how I can do this behavior with Next.js?

Next.js version: ^12.2.2
React version: ^18.2.0"
Typescript: 4.6.2

**Comment by leerob** [comment]
**Update:** We’re excited to announce [Next.js 13](https://nextjs.org/13), which includes the new `app/` directory (beta) with support for nested layouts, colocation of data fetching, streaming, Server Components by default, and much more. Further, you can use packages that import their own .css files without additional configuration.

Inside the `app/` directory, CSS files can now be imported inside any layout or page, including styles from external npm packages. This solves the original proposal in this RFC.
